file/fs.lisp

```diff
@@ -71,18 +82,36 @@
                (error 'unknown-host
                       :host name
                       :pathname (format nil "~A:" name))))))
-    (t name)))
-
-(defun (setf find-host) (new-value name &optional errorp)
-  (declare (ignore errorp))
+    (file-system-host name)
+    (t (when errorp
+         (error 'mezzano.internals::simple-type-error
+                :expected-type '(or string symbol file-system-host)
+                :datum name
+                :format-control "~S is invalid type ~S, expected one of ~S"
+                :format-arguments (list name
+                                        (type-of name)
+                                        '(string symbol file-system-host)))))))
+
+(defun (setf find-host) (new-value name &optional (errorp t))
   (setf name (string-upcase (string name)))
   (assert (not (zerop (length name))))
-  (cond (new-value
+  (cond ((null new-value)
+         (setf *host-alist* (remove name *host-alist* :key 'first :test 'string=))
+         NIL)
+        ((typep new-value 'file-system-host)
          (setf *host-alist*
                (list* (list name new-value)
-                      (remove name *host-alist* :key 'first :test 'string=))))
-        (t (setf *host-alist* (remove name *host-alist* :key 'first :test 'string=))))
-  new-value)
+                      (remove name *host-alist* :key 'first :test 'string=)))
+         new-value)
+        (errorp
```

Comment:
Same thing here, except that errorp only exists so that (setf find-host)'s lambda-list mirrors find-host's.
There should be a (check-type new-value (or null file-system-host)) at the start instead.